src/topology/topological_fiber_bundle.lean

```diff
@@ -0,0 +1,476 @@
+situation: even if Lean can not check that two basepoints are defeq, it will accept the fact that
+the tangent spaces are the same. For instance, if two maps f and g are locally inverse to each
+other, one can express that the composition of their derivatives is the identity of
+`tangent_space 𝕜 x`. One could fear issues as this composition goes from `tangent_space 𝕜 x` to
+`tangent_space 𝕜 (g (f x))` (which should be the same, but should not be obvious to Lean
+as it does not know that `g (f x) = x`). As these types are the same to Lean (equal to `F`), there
+are in fact no dependent type difficulties here!
+
+For this construction, we should thus choose for each `x` one specific trivialization around it. We
+include this choice in the definition of the fiber bundle, as it makes some constructions more
+functorial and it is a nice way to say that the trivializations cover the whole space B.
+
+With this definition, the type of the fiber bundle is just `Σ x : B, F`, but the topology is not the
+product one.
+
+We also take the indexing type (indexing all the trivializations) as a parameter to the bundle:
+it could always be taken as a subtype of all the maps from open subsets of B to continuous maps
+of F, but in practice it will sometimes be something else. For instance, on a manifold, one will use
+the set of charts as a good parameterization for the trivializations of the tangent bundle. Or for
+the pullback of a fiber bundle the indexing type will be the same as for the initial bundle.
+
+## Tags
+Fiber bundle, topological bundle, vector bundle, local trivialization, structure group
+-/
+
+variables {ι : Type*} {B : Type*} {F : Type*}
+
+open topological_space set
+
+section topological_fiber_bundle
+
+variables {Z : Type*} [topological_space B] [topological_space Z]
+  [topological_space F] (proj : Z → B)
+
+variable (F)
+
+/-- A structure extending local homeomorphisms, defining a local trivialization of a projection
+`proj : Z → B` with fiber `F`, as a local homeomorphism between `Z` and `B × F`. -/
+structure bundle_trivialization extends local_homeomorph Z (B × F) :=
+(base_set      : set B)
+(open_base_set : is_open base_set)
+(source_eq     : source = proj ⁻¹' base_set)
+(target_eq     : target = set.prod base_set univ)
+(proj_to_fun   : ∀ p ∈ source, (to_fun p).1 = proj p)
+
+/-- A topological fiber bundle with fiber F over a base B is a space projecting on B for which the
+fibers are all homeomorphic to F, such that the local situation around each point is a direct
+product. -/
+def is_topological_fiber_bundle : Prop :=
+∀ x : Z, ∃e : bundle_trivialization F proj, x ∈ e.source
+
+variables {F} {proj}
+
+/-- In the domain of a bundle trivialization, the projection is continuous-/
+lemma bundle_trivialization.continuous_at_proj (e : bundle_trivialization F proj) {x : Z}
+  (ex : x ∈ e.source) : continuous_at proj x :=
+begin
+  assume s hs,
+  rw mem_nhds_sets_iff at hs,
+  rcases hs with ⟨t, st, t_open, xt⟩,
```

Comment:
For your readability toolbox: do you know you can replace the above two lines by:
```lean
  obtain ⟨t, st, t_open, xt⟩ : ∃ t ⊆ s, is_open t ∧ proj x ∈ t,
    from mem_nhds_sets_iff.1 hs,
```